crates/astria-conductor/src/data_availability.rs

```diff
@@ -330,41 +352,61 @@ async fn verify_sequencer_blobs_and_assemble_rollups(
-/// against the sequencer "header" blob.
-async fn fetch_and_verify_rollup_blobs(
-    client: CelestiaClient,
-    height: u64,
+/// Fetches a rollup blob for given height and namespace, and fowards it to the assembler.
+///
+/// If more than one rollup blob is received and pass verification, they are all dropped.
+/// It is assumed that sequencer-relayer submits at most one rollup blob to celestia per
+/// celestia height.
+#[instrument(skip_all, fields(height, block_hash = %data.data().block_hash))]
+async fn fetch_verify_rollup_blob_and_forward_to_assembly(
+    client: HttpClient,
+    height: Height,
     data: SignedNamespaceData<SequencerNamespaceData>,
     namespace: Namespace,
     block_tx: mpsc::Sender<SequencerBlockSubset>,
 ) {
-    let rollup_data = match client.get_rollup_data(height, &data, namespace).await {
+    let mut rollups = match client
+        .get_rollup_data_matching_sequencer_data(height, namespace, &data)
+        .await
+    {
         Err(e) => {
             warn!(error.message = %e, error.cause = ?e, "failed to get rollup data from celestia");
             return;
         }
-        Ok(None) => {
+        Ok(rollups) if rollups.is_empty() => {
             info!("could not find rollup data");
             return;
         }
-        Ok(Some(rollup_data)) => rollup_data,
+        Ok(rollups) => rollups,
     };
 
-    if block_verifier::validate_rollup_data(&rollup_data, data.data.action_tree_root).is_err() {
-        warn!(
-            "could not validate rollup data given the action tree root of the sequencer data; \
-             dropping the block"
-        )
-    } else {
-        let subset = SequencerBlockSubset {
-            block_hash: data.data.block_hash,
-            header: data.data.header,
-            rollup_transactions: rollup_data.rollup_txs,
-        };
-        if block_tx.send(subset).await.is_err() {
-            warn!("failed sending validated rollup data to block assembler; receiver dropped");
+    let num_rollups_received = rollups.len();
+    info!(
+        rollups.received = num_rollups_received,
+        "received rollups; verifying"
+    );
+    rollups.retain(|rollup| {
+        block_verifier::validate_rollup_data(rollup, data.data().action_tree_root).is_ok()
+    });
+    match rollups.len() {
+        0 => info!("all received rollups failed verification; not forwarding to assembler"),
+        1 => {
+            info!("one rollup successfully verified; forwarding to block assembler");
+            let subset = SequencerBlockSubset {
+                block_hash: data.data().block_hash,
+                header: data.data().header.clone(),
+                rollup_transactions: rollups.swap_remove(0).rollup_txs,
+            };
+            if block_tx.send(subset).await.is_err() {
+                warn!("failed sending validated rollup data to block assembler; receiver dropped");
+            }
         }
+        num_rollups_verified => info!(
```

Comment:
maybe change to warn? this would require investigation if it happened

Reply:
ty, good catch. That's what I wanted